libs/libdsp/lib_avg.c

```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+ * libs/libdsp/lib_avg.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <assert.h>
+#include <dsp.h>
+#include <string.h>
+
+/* Based on video explanation of Dr. Shane Ross:
+ * https://www.youtube.com/watch?v=HCd-leV8OkU
+ */
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: avg_filter_data_init
+ *
+ * Description:
+ *   Initialize the data struct used to store prev_avg and k parameter.
+ *
+ * Input Parameters:
+ *   data     - pointer to avg_filter_data_s
+ *   prev_avg - initial value of prev_avg
+ *   k        - initial value of k counter
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void avg_filter_data_init(FAR struct avg_filter_data_s *data,
```

Comment:
This API seems to be redundant since does nothing than simple assignment. Since context structure is exposed to user the explicit field assignment can be done by the caller. If we need a kind of init, maybe macro can be used?

Reply:
This is just a suggestion. We can leave the API

Reply:
I think it would be good if zero initialized structure can be a default.

Reply:
It was @raiden00pl suggestion, I think the user could access the struct fields directly, but it is done this way for other DSP functions.

Reply:
Yes. Sure.

Reply:
Just to offer an alternative view, an API to initialize can be more future-proof than requiring consumers to "manually" initialize.